### ISSUE TITLE
Cubic spline - x labels after fitting

### ISSUE THREAD

**Issue by Amarounek** [altair-viz/altair#2326]
Dear all,
 
I have an issue with my cubic spline figure.  As you can see, I would like to fit my Cubic spline to the figure with Linear regression and scatterplot however, I have an issue with visualization some reasonable x-axis. What should I do better to provide a reasonable x-axis (in years)? Also, a minor issue, but how to put Linear regression to the legend?

![visualization (1)](https://user-images.githubusercontent.com/36733547/97110798-7fa37600-16db-11eb-99c8-ebd810e218ec.png)

If I change the x-axis values to **Nominal**
![image](https://user-images.githubusercontent.com/36733547/97111164-89c67400-16dd-11eb-8944-ecfedc4e7f3a.png)

Here is my code snippet:
```
#packages
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from patsy import dmatrix
import statsmodels.api as sm
import altair as alt
from altair import Chart, Color, Scale

rng = np.random.RandomState(1)
x = np.random.uniform(low=40, high=365, size=(36))
df = pd.DataFrame({'Number of rainy days': x, 'Year': range(1984, 2020)})
df['ID'] = prediction_df.index
df.tail(n=1)

# Linear regression
y = df['Number of rainy days']
x = df['Year']
x = x.values.reshape(-1,1)
y = y.values.reshape(-1,1)
lm = LinearRegression()
lm.fit(x , y)
prediction = lm.predict(x)
prediction_df = pd.DataFrame(prediction)
prediction_df = prediction_df.rename(columns={0: "Linear regression"}) 
prediction_df['Year'] = range(1984, 2020)

# Specifying 4 degrees of freedom (Cubic spline)
transformed_x3_wet = dmatrix("cr(df.Year, df=4)", {"df.Year": df.ID}, return_type='dataframe')
fit_wet = sm.GLM(df['Number of rainy days'], transformed_x3_wet).fit()
ID_grid = np.arange(df.Year.min(), df.Year.max()).reshape(-1,1)
pred_wet_days = fit_dry_3.predict(dmatrix("cr(ID_grid , df=4)", {"ID_grid ": ID_grid }, return_type='dataframe'))
wet_days_df = pd.DataFrame({'Cubic spline': pred_wet_days, 'xfit': np.linspace(df['Year'].min(), df['Year'].max(), 35)})

# Plot the data points on an interactive axis
points = alt.Chart(df).mark_circle(color='black').encode(
    x=alt.X('Year:Q', title='Year', axis=alt.Axis(titlePadding=10)),
    y=alt.Y('Number of rainy days', title='Number of rainy days', axis=alt.Axis(titlePadding=10))
)
# Plot the Linear regression
bar_linear = alt.Chart(prediction_df).mark_line(color='green').encode(
    alt.X('Year:Q'),
    alt.Y('Linear regression'))

# Plot the best fit polynomials
polynomial_fit = alt.Chart(wet_days_df).transform_fold(
    ['Cubic spline'],
    as_=['degree', 'yfit']
).mark_line().encode(
    x= alt.X('xfit:Q'),
    y= alt.Y('yfit:Q'),
    color = alt.Color('degree:N', title = 'Type of regression')
)

bar_linear.properties(width=700, height=400) + points + polynomial_fit

```
Thanks for any advice!

**Comment by jakevdp** [comment]
You can pass `format='d'` in the `alt.Axis` to display the years as integers without commas.

**Comment by Amarounek** [comment]
Thanks! I think closed! ;-)
